virtual_rainforest/__init__.py

```diff
@@ -1,13 +1,18 @@
 import importlib.metadata
+import pkgutil
+from importlib import import_module
 
-# Import all module schema here to ensure that they are added to the registry
-from virtual_rainforest.core import schema  # noqa
-from virtual_rainforest.models.abiotic import schema  # noqa
-from virtual_rainforest.models.abiotic.abiotic_model import AbioticModel  # noqa
-from virtual_rainforest.models.plants import schema  # noqa
-from virtual_rainforest.models.soil import schema  # noqa
+import virtual_rainforest.models as vfm
 
-# Import models here so that they also end up in the registry
-from virtual_rainforest.models.soil.soil_model import SoilModel  # noqa
+# Import the core schema to register it
+from virtual_rainforest.core import schema  # noqa: F401
+
+# Autodiscover models in the models module to add their schema and BaseModel subclass
+# All modules within virtual_rainforest.model are expected to:
+# - import their BaseModel subclass to the module root, for example:
+#   from virtual_rainforest.models.soil.model import SoilModel  # noqa: F401
+# - define a schema member, decorated with @register_schema
+for module_info in pkgutil.iter_modules(vfm.__path__):
+    import_module(f"virtual_rainforest.models.{module_info.name}")
```

Comment:
...alternatively, you could check here whether the module has an attribute called `schema` and if not, add a default one.

Reply:
One thing that crossed my mind was having a `config.add_module` command which basically wraps this import and schema registration. That could then be used to add custom modules _outside_ of the package tree. If you're at the Python command line, you can always import them, but if you wanted to use them in a config, we'd need a `custom_model` section that could use this too. But, that is probably one for the future.